### DISCUSSION TITLE
How to stop a loop

### DISCUSSION THREAD

**Discussion by bayeroo22** [autokey/autokey#930] [Q&A] [answered]
Hello, 
I have a simple script and I would like to stop or interrupt it when I want to:
I start the script with Hotkey F1 and i want to stop it with F2 (maybe after 4 loops).
 
-----------------------------------------------------------------------------------------------------------------------
import time

def repeat_input():
    count = 0
    while count < 10:  # loop 10 times
        #keyboard.wait('0')
        keyboard.send_keys("1")
        time.sleep(0.1)
        keyboard.send_keys("2")
        time.sleep(0.1)
        keyboard.send_keys("3")
        time.sleep(0.1)
        keyboard.send_keys("4")
        time.sleep(0.1)
        keyboard.send_keys("5")
        time.sleep(5.0)
        
        count += 1

# Start
repeat_input()
---------------------------------------------------------------------------------------------------------------------------

**Comment by josephj11** [comment, marked as answer]
Welcome to the AutoKey community, @bayeroo22 !

[This](https://github.com/autokey/autokey/wiki/Contributed-Scripts-2#toggle-a-looped-script) should do it for you.

Be sure to check out the rest of the wiki. There are a lot of helpful articles and sample scripts.

Joe

FYI: The time module is already exposed to all AutoKey scripts, so you don't need to import it, but Python is smart, so doing so doesn't hurt anything.
